kernel/src/tdx/tdcall.rs

```diff
@@ -0,0 +1,234 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+//
+// Copyright (c) Microsoft Corporation
+//
+// Author: Jon Lange <jlange@microsoft.com>
+
+use super::error::{tdx_result, TdxError};
+use crate::address::{Address, PhysAddr, VirtAddr};
+use crate::error::SvsmError;
+use crate::mm::pagetable::PageFrame;
+use crate::mm::{virt_to_frame, PerCPUPageMappingGuard};
+use crate::types::{PAGE_SHIFT, PAGE_SIZE, PAGE_SIZE_2M};
+use crate::utils::MemoryRegion;
+
+use core::arch::asm;
+
+const TDG_VP_TDVMCALL: u32 = 0;
+const TDCALL_TDG_MEM_PAGE_ACCEPT: u32 = 6;
+
+const TDVMCALL_HLT: u32 = 12;
+const TDVMCALL_IO: u32 = 30;
+
+fn gpa_info(frame: PageFrame) -> u64 {
+    // Callers are expected to guarantee alignment before calling this
+    // function.  If the caller verifies the alignment, then it can be done
+    // once without having to do it in a loop.
+    debug_assert!(u64::from(frame.address()) & (frame.size() - 1) as u64 == 0);
```

Comment:
```suggestion
    debug_assert_eq!(u64::from(frame.address()) & (frame.size() - 1) as u64, 0);
```
If the value is not `0`, it'd be useful to know what it was.

What's the reasoning behind using `debug_assert!` rather than just `assert!`. Is see that we're already checking the alignment in `td_accept_physical_memory` and `td_accept_virtual_memory`, but we don't in `td_accept_virtual_4k` and `td_accept_virtual_2m`.